### ISSUE TITLE
UDP is not shown in DP-API-Gateway after creating gateway extension

### ISSUE THREAD

**Issue by ragkna** [ozairs/apiconnect-policies#1]
Hi Ozair,
I followed the instruction given but after creating gateway extension I can not see user-defined-policies in the drop down (See attachment).

Also noticed that before disabling api connect gateway service "Gateway peering manager" was set to "none" but after disabling it the value set to "default" and its status went down. (See attachment).
Have you experienced this?

![image](https://user-images.githubusercontent.com/20474360/64148028-892da700-ce65-11e9-9cdd-d6b6950faa82.png)

**Comment by ozairs** [comment]
In the LTE, when using Gateway Peering Manager, you should leave it as `none`. Since your option does not appear greyed out, then there could be an issue in the peering, but for development purposes, try switching it off to focus your troubleshooting.

Since the UDP drop-down is empty, then the objects are not created in the gateway. Review the gateway extensions cli response, or take the packaged zip and import it manually via DataPower Import Configuration.

**Comment by ragkna** [comment]
Hi Ozair,

I verified outputs of apic gateway-extensions responses and they are same as you mentioned. 
I manually created assembly function in DP and attached it to apic-gw-service which then showed me in API designer.

![image](https://user-images.githubusercontent.com/20474360/64216000-626d8000-cefa-11e9-9dcc-3cac9464a929.png)

But if I do a get on gateway-extensions it did not show any.
_**_apic gateway-extensions:get --scope org -o admin --gateway-service datapower-api-gateway --availability-zone availability-zone-default --server 127.0.0.1:2000 --output -
Error: Resource not found_**_

This could be a problem when we have multiple DPs in the environment. We are after APIm pushing extensions to all clustered DPs and not the case where we need to deploy these policies to DPs individually.

**Comment by ozairs** [comment]
in your `apic gateway-extensions` command, you need to modify the `server` flag to reflect your management address instead of `127.0.0.1:2000`

**Comment by ragkna** [comment]
It did not make any difference. I will try this on client environment.

Another question: Can we use xml files in custom policy (used to store property mapping)? If yes, how do we package this to upload to APIm gateway extension?

**Comment by ozairs** [comment]
yes, just package files into gateway extensions. See repo for examples on specifying the appropriate path based on file system placement.

**Comment by ragkna** [comment]
Hi Ozair,
I gave a try at a customer (on-prem) and no luck so far. I feel that there is something wrong in the command.

For example: 
This works
apic gateway-extensions:create ./gw-extension.zip --scope org --org **admin** --gateway-service apic-sandpit --availability-zone application-availability-zone-1 --server apic.internal

This does not work: (replaced --org param value with actual organisation name created in cloud manager)
apic gateway-extensions:create ./gw-extension.zip --scope org --org **myorg** --gateway-service apic-sandpit --availability-zone application-availability-zone-1 --server apic.internal

I get this error: **Error: Resource not found**

Is the org param value always hard coded to "admin"?

Also, I noticed that when I tried to upload using the first command, I see few events are recorded in DP system logs, but not errors. An again, if I look for any user defined policies or assembly function in the domain, none exists!

**Comment by ozairs** [comment]
> Is the org param value always hard coded to "admin"?

yes

**Comment by mtorresVcsoft** [comment]
H Ozairsi,
I managed to successfully implement UDP, but the APIM interface does not show me the created UDP, I have DataPower API Gateway and API Connect V 2018.4.1.9.

![image](https://user-images.githubusercontent.com/48832966/77709675-da37f080-6f99-11ea-8328-35f7b6f1bafa.png)

![image](https://user-images.githubusercontent.com/48832966/77709713-f2a80b00-6f99-11ea-858b-5085f5c4688a.png)



Thank you.
